### PR TITLE
ResStock-HPXML: Upgrade costs, Retain HVAC sizes

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

(1) Introduce new `UpgradeCosts` reporting measure, which is essentially the upgrade portion of the old `SimulationOutputReport` measure. We now retrieve cost multipliers directly from the `hpxml` instead of the `model`.

(2) Include logic in `ApplyUpgrade` for determining whether to retain calculated HVAC capacities or not. In other words, checks if any options include upgrades to the heating system, supplemental heating system, cooling system, or heat pump; for each that exists, we don't retain that system's capacity.

The `UpgradeCosts` measure:
* has no arguments
* gets cost multipliers from `in.xml`, the defaulted version of `existing.xml` (the exported hpxml from `BuildExistingModel`)
* multiplies cost value (saved by `ApplyUpgrade`) by cost multiplier value
* has unit tests that operate on test osw files corresponding to test files for the old `SimulationOutputReport` measure

Additionally:
* change cost multiplier `Duct Surface Area (ft^2)` to `Duct Unconditioned Surface Area (ft^2)`
* split cost multiplier `Size, Heating Supplemental System (kBtu/h)` into separate cost multipliers:
  * `Size, Heating Backup System (kBtu/h)` (ASHP backup)
  * `Size, Heating Secondary System (kBtu/h)` (separate secondary heating)
* add `Flow Rate, Mechanical Ventilation (cfm)` cost multiplier
* add `Slab Perimeter, Exposed, Conditioned (ft)` cost multiplier 
* `BuildExistingModel` exports `existing.osw` and `ApplyUpgrade` exports `upgraded.osw` (instead of `measures.osw` and `measures-upgrade.osw`)
* more test workflows (HVAC-related, non HVAC-related, package with both) for `test:measures_osw` rake task

**Test Results**
Ran `integration:workflow_tests` rake task with `n_samples=10` and [1 base + 9 upgrades](https://github.com/NREL/resstock/tree/restructure-v3-costs-and-sizes/test/test_samples_osw) for `project_testing`:
* [baseline](https://github.com/NREL/resstock/files/5966232/results.csv.txt)
* [upgrades](https://github.com/NREL/resstock/files/5966222/results.csv.txt)

There are no datapoint failures.

Update. Here is a baseline run with `n_samples=100`:
* [project_national](https://github.com/NREL/resstock/files/5966218/results.csv.txt)

Again, no datapoint failures.

## Checklist

Not all may apply:

- [ ] Unit tests have been added or updated
- [ ] All rake tasks have been run, and pass
- [ ] Documentation has been modified appropriately
- [ ] Any new options are added to `project_testing`
- [ ] `project_testing` runs without any failures
- [ ] No unexpected regression test changes
- [ ] All tests are passing
- [ ] The [changelog](https://github.com/NREL/resstock/blob/master/CHANGELOG.md) has been updated appropriately
- [ ] This branch is up-to-date with develop

For more information on how to perform these checklist items, see the documentation's [Advanced Tutorial](https://resstock.readthedocs.io/en/latest/advanced_tutorial/index.html).

### CODE REVIEW

File: measures/ApplyUpgrade/measure.rb [hunk 1]
```diff
@@ -281,6 +279,34 @@ def run(model, runner, user_arguments)
...
+          args_hash.each do |arg, value|
+            if arg.include?('heating_system_type') || arg.include?('heating_system_fuel') || arg.include?('heating_system_heating_efficiency') || arg.include?('heating_system_fraction_heat_load_served')
+              upgrade_heating_system = true
+            end
+          end
+
+          # Detect whether we are upgrading the supplemental heating system
+          args_hash.each do |arg, value|
+            if arg.include?('heating_system_type_2') || arg.include?('heating_system_fuel_2') || arg.include?('heating_system_heating_efficiency_2') || arg.include?('heating_system_fraction_heat_load_served_2')
+              upgrade_suppy_heating_system = true
```
Comment (shorowit) [inline measures/ApplyUpgrade/measure.rb:292]: Typo

Comment (joseph-robertson) [reply measures/ApplyUpgrade/measure.rb:292]: @shorowit Overall what do you think of this approach?

Comment (shorowit) [reply measures/ApplyUpgrade/measure.rb:292]: I think it makes sense. Would be nice to have some tests that demonstrate it's working for different HVAC systems types. (Such a test would catch the above typo, for example! 😉)

**Review by shorowit — COMMENTED** [review]
One more thought... can you somehow create tests for the ApplyUpgrade measure that show A) HVAC cost multipliers are updated when the upgrade includes an HVAC change and B) HVAC costs are _not_ updated when the upgrade excludes an HVAC change?

**Comment by joseph-robertson** [comment]
@shorowit Thanks for the review. I tried to address all of your comments, including taking a crack at creating tests for the ApplyUpgrade measure.

**Review by shorowit — APPROVED** [review]
Looks good to me. Nice changes based on my feedback.

**Pull request merged**